src/components/NewCommentForm.jsx

```diff
@@ -0,0 +1,87 @@
+import { useContext, useState } from 'react';
+import { postNewComment } from '../utils/api';
+import { UserContext } from '../contexts/User';
+
+export const NewCommentForm = ({ article_id, setComments }) => {
+  const { user, setUser } = useContext(UserContext);
+
+  const [newComment, setNewComment] = useState({
+    username: user,
+    body: '',
+  });
+  const [badInput, setBadInput] = useState(false);
+  const [isPostError, setIsPostError] = useState(false);
+
```

Comment:
A new piece of state will help you manage someone trying to post the same comment multiple times (smashing the submit button!)